lib/spacesuit/auth_middleware.ex

```diff
@@ -29,7 +29,39 @@ defmodule Spacesuit.AuthMiddleware do
     end
   end
 
+  defp handle_bearer_token(req, env, token) do
+    if bypass_session_srv?(env) do
+      case session_service()[:impl].validate_api_token(token) do
+        :ok ->
+          {:ok, req, env}
+        unexpected -> # Otherwise we blow up the request
+          Logger.error "auth_middleware error: unexpected response - #{inspect(unexpected)}"
+          error_reply(req, 401, "Bad Authentication Token")
+          {:stop, req}
+      end
+    else
+      session_service()[:impl].handle_bearer_token(req, env, token, session_service()[:url])
+    end
+  end
+
   defp strip_auth(req) do
     %{ req | headers: Map.delete(req[:headers], "authorization") }
   end
+
+  # should the session service be bypassed for this route?
+  defp bypass_session_srv?(env) do
+    get_in(env, [:handler_opts, :middlewares_config, :auth, :bypass_session_srv]) || false
```

Comment:
@relistan the intension here is to allow configuration like this:
```
    #All /v1/users[...] -> usersrv
    { "/v1/users/[...]", %{
      description: "user service - users",
      all_actions: "#{user_svc}/users/[...]",
      add_headers: %{
        "X-Forwarded-Prefix" => "/v1"
      },
      middlewares_config: %{
        :auth => %{
          :bypass_session_srv => true
        }
      }
    }},
```
is this what you had in mind with [your previous comments](https://github.com/relistan/spacesuit/pull/8#discussion_r130445318)? I'm not partial to the keys defined here but its a start for discussion purposes. Conceivably, we could also rewrite the CORS middleware on a route-by-route basis as well.